### PR TITLE
dependentKey common validator options

### PR DESCRIPTION
**Pull request by offirgolan**
For #27

### CODE REVIEW

**Comment by offirgolan** [comment]
ready for review :smiley_cat:

File: tests/unit/build-validation-test.js [hunk 1]
```diff
@@ -30,28 +30,28 @@
 
  var container, registry;
 
-function register(fullName, factory) {
-  (registry || container).register(fullName, factory);
-}
+ function register(fullName, factory) {
```
Comment (rwjblue) [inline tests/unit/build-validation-test.js]: Why was leading whitespace added in front of these lines?

Comment (offirgolan) [inline tests/unit/build-validation-test.js]: woah no idea. Let me fix that

**Comment by rwjblue** [comment]
I left one small comment, but looks good other than that.

In the future, could we keep the auto-generated doc updates in a separate commit than the main code and docs changes?  It makes reading the diff somewhat more difficult (honestly, it would be nice if we could avoid checking in the generated files but that is a whole other issue).

**Pull request merged**